src/main/java/com/exadel/discount/filters/JwtFilter.java

```diff
@@ -0,0 +1,105 @@
+package com.exadel.discount.filters;
+
+import com.exadel.discount.config.JwtVariablesConfig;
+import com.exadel.discount.service.UserDetailsServiceImpl;
+import com.exadel.discount.util.jwt.JwtUtil;
+import com.exadel.discount.util.jwt.RefreshJwtUtil;
+import lombok.RequiredArgsConstructor;
+import lombok.Setter;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.security.authentication.AbstractAuthenticationToken;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.web.authentication.WebAuthenticationDetailsSource;
+import org.springframework.security.web.authentication.preauth.PreAuthenticatedAuthenticationToken;
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+import static com.exadel.discount.config.JwtVariablesConfig.REFRESH_ROLE;
+
+@Component
+@Setter
+@RequiredArgsConstructor
+public class JwtFilter extends OncePerRequestFilter {
+    private final String AUTHORIZATION_HEADER = "Authorization";
+    private final String BEARER_TYPE_OF_AUTHORIZATION_HEADER = "Bearer ";
+
+    private final UserDetailsServiceImpl userDetailsService;
+
+    private AbstractAuthenticationToken authentication;
```

Comment:
Don't use class fields for temp usage. It will be broken on multiple parallel requests